bpmn-statistic/src/com/axonivy/utils/bpmnstatistic/utils/ProcessesMonitorUtils.java

```diff
@@ -228,11 +235,112 @@ private static void followPath(AlternativePath path, SequenceFlow currentFlow) {
+    return CustomFieldType.STRING == type || CustomFieldType.TEXT == type;
+  }
+
+  private static List<Object> flattenStringCustomFieldValues(List<Object> customFieldValues) {
+    return customFieldValues.stream().filter(value -> value instanceof String[]).map(value -> (String[]) value)
+        .flatMap(Arrays::stream).collect(Collectors.toList());
+  }
+
+  /**
+   * Cast values to right type and Create sub-query for each custom field type from case or task
+   **/
+  @SuppressWarnings("unchecked")
+  private static void addCustomFieldSubQuery(CaseQuery customFieldQuery, CustomFieldFilter customFieldFilter,
+      Object customFieldValue) {
+    boolean isCustomFieldFromCase = customFieldFilter.isCustomFieldFromCase();
+    String customFieldName = customFieldFilter.getCustomFieldMeta().name();
+    CustomFieldType customFieldType = customFieldFilter.getCustomFieldMeta().type();
+
+    switch (customFieldType) {
+      case STRING:
+        String stringValue = (String) customFieldValue;
+        if (isCustomFieldFromCase) {
+          customFieldQuery.where().or().customField().stringField(customFieldName).isEqual(stringValue);
+        } else {
+          customFieldQuery.where().or()
+              .tasks(TaskQuery.create().where().customField().stringField(customFieldName).isEqual(stringValue));
+        }
+        break;
+      case TEXT:
+        String textValue = (String) customFieldValue;
+        if (isCustomFieldFromCase) {
+          customFieldQuery.where().or().customField().textField(customFieldName).isEqual(textValue);
+        } else {
+          customFieldQuery.where().or()
+              .tasks(TaskQuery.create().where().customField().textField(customFieldName).isEqual(textValue));
+        }
+        break;
+      case NUMBER:
+        List<String> numberRange = (List<String>) customFieldValue;
+
+        Double startNumber = Double.parseDouble(numberRange.get(0));
+        Double endNumber = Double.parseDouble(numberRange.get(1));
+
+        if (isCustomFieldFromCase) {
+          customFieldQuery.where().or().customField().numberField(customFieldName).isGreaterOrEqualThan(startNumber)
+              .and().customField().numberField(customFieldName).isLowerOrEqualThan(endNumber);
+        } else {
+          customFieldQuery.where().or()
+              .tasks(TaskQuery.create().where().customField().numberField(customFieldName)
+                  .isGreaterOrEqualThan(startNumber).and().customField().numberField(customFieldName)
+                  .isLowerOrEqualThan(endNumber));
+        }
+        break;
+      case TIMESTAMP:
+        List<LocalDate> dateRange = (List<LocalDate>) customFieldValue;
+        LocalDate startDate = dateRange.get(0);
+        LocalDate endDate = dateRange.get(1);
+
+        if (isCustomFieldFromCase) {
+          customFieldQuery.where().or().customField().timestampField(customFieldName)
+              .isGreaterOrEqualThan(java.sql.Date.valueOf(startDate)).and().customField()
```

Comment:
Disclaimer: For illustrating existing java apis only. Should not be used in production code.
You can use java.sql.Date.valueOf() method
Please adapt this one